vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/resources/META-INF/resources/frontend/datepickerConnector.js

```diff
@@ -152,9 +158,14 @@ import { extractDateParts, parseDate as _parseDate } from '@vaadin/date-picker/s
           const formatterAndParser = createFormatterAndParser(usedFormats);
 
           // Merge current web component I18N settings with new I18N settings and the formatting and parsing functions
-          // Defer setting I18N property into microtask as a workaround for: https://github.com/vaadin/flow-components/issues/4500
-          queueMicrotask(() => {
-            datepicker.i18n = Object.assign({}, datepicker.i18n, i18n, formatterAndParser);
+          const updatedI18n = Object.assign({}, datepicker.i18n, i18n, formatterAndParser);
+          datepicker.i18n = updatedI18n;
+          // Update I18N property again after the element is ready as a workaround for both:
+          // https://github.com/vaadin/flow-components/issues/4500
+          // https://github.com/vaadin/flow-components/issues/4667
+          // This workaround is only necessary for v23, and will be removed in v24.
+          datepicker.addEventListener("date-picker-ready", () => {
+            requestAnimationFrame(() => datepicker.i18n = updatedI18n);
```

Comment:
Not sure if related to this PR, but if the root items are initially collapsed and then expanded, the time pickers have the wrong locale:

https://user-images.githubusercontent.com/1222264/220933193-7ee74951-50fe-43a4-bcce-315027ef2f8a.mp4

Reply:
Can be related to https://github.com/vaadin/flow-components/pull/4706/files#r1115759943

Reply:
I rolled back the change and it is still reproducible.

Reply:
That part is still reproducible with the latest updates. Maybe we can separate that issue. What do you think?

Reply:
Is it reproducible on 23.3 without these changes?

Reply:
I checked again, and it is reproducible. I also checked how it works on v24, and it shows "12:00 PM" for both root rows and the expanded rows. I will verify if that is the expected behavior. If so, the time pickers in the root row are broken in v23.3, otherwise the ones in the children rows are broken along with the time pickers in v24.